File: pset1/cash/cash.c
// Program to get the minimum number of coins for certain change
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // No negative cash/change
    float f = -0.1;
    while (f < 0)
    {
        f = get_float("Change owed: ");
    }

    // Conversion to int for calculation
    int change = (int)(roundf(f * 100));

    // Let's make it complicated...
    int quarter = 25, dime = 10, nickel = 5, penny = 1;
    int n = 0;
    while (change > 0)
    {
        change -= change >= quarter ? quarter : (change >= dime ? dime : (change >= nickel ? nickel : (change >= penny ? penny : 0)));
        n++;
    }
    printf("%d\n", n);
}

File: pset6/cash/cash.py
# Program to get the minimum number of coins for certain change
from cs50 import get_float

while True:
    f = get_float("Change owed: ")
    # No negative cash/change
    if f >= 0:
        break

change = int(f * 100)

# Get number of coins
quarter, dime, nickel, penny, n = 25, 10, 5, 1, 0
while change > 0:
    change -= quarter if change >= quarter else (
        dime if change >= dime else (
            nickel if change >= nickel else (
                penny if change >= penny else 0)))
    n += 1
print(n)